pkg/error.go

```diff
@@ -0,0 +1,8 @@
+package pkg
+
+// Error type is immutable error type
+type Error string
```

Comment:
I'm not sure we need this new type.
Could we just use the standard error and the errors package to create them i.e. ```errors.New("variation not found")```